Failing tests in CBC and SDR due to floating point precision

## Description
I’ve been seeing failures in two regression tests on GitHub Actions that look like floating point precision errors (see #2252). These are potentially just intermittent issues related to runner architecture, but it could be helpful to adjust the tolerance so this doesn't keep happening.

The failing tests are:

- `tests/test_coastal_blue_carbon.py::TestCBC2::test_model`
- `tests/test_sdr.py::SDRTests::test_base_regression`

When they fail, the difference values are extremely small:


```=========================== short test summary info ============================
FAILED tests/test_coastal_blue_carbon.py::TestCBC2::test_model - AssertionError: 
Not equal to tolerance rtol=1e-06, atol=0

Mismatched elements: 1 / 2 (50%)
Max absolute difference among violations: 3.8146973e-05
Max relative difference among violations: 1.909334e-06
 ACTUAL: array([[-19.97924 ,  44.801987]], dtype=float32)
 DESIRED: array([[-19.979202,  44.801994]], dtype=float32)
FAILED tests/test_sdr.py::SDRTests::test_base_regression - AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

Mismatched elements: 1 / 1 (100%)
Max absolute difference among violations: 4.2293172e-08
Max relative difference among violations: 1.0846527e-07
 ACTUAL: array(0.389924)
 DESIRED: array(0.389924)
= 2 failed, 679 passed, 10 skipped, 44 warnings, 4 subtests passed in 110.98s (0:01:50) =
make: *** [Makefile:161: test] Error 1
Error: Process completed with exit code 2.```

--- a/tests/test_coastal_blue_carbon.py
+++ b/tests/test_coastal_blue_carbon.py
@@ -739,7 +739,7 @@ class TestCBC2(unittest.TestCase):
             raster = gdal.OpenEx(raster_path)
             numpy.testing.assert_allclose(
                 raster.ReadAsArray(),
-                expected_total_sequestration, rtol=1e-6)
+                expected_total_sequestration, rtol=1e-5)
         finally:
             raster = None
 

--- a/tests/test_sdr.py
+++ b/tests/test_sdr.py
@@ -190,7 +190,7 @@ class SDRTests(unittest.TestCase):
                 raster_sum += numpy.sum(
                     block[~pygeoprocessing.array_equals_nodata(
                             block, nodata)], dtype=numpy.float64)
-            numpy.testing.assert_allclose(raster_sum, expected_sum)
+            numpy.testing.assert_allclose(raster_sum, expected_sum, atol=1e-5)
 
     def test_base_regression_d8(self):
         """SDR base regression test on sample data in D8 mode.

